staging/src/k8s.io/component-base/metrics/prometheusextension/timing_histogram.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prometheusextension
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+
+	"k8s.io/utils/clock"
+)
+
+// GaugeOps is the operations a user of a gauge can do to the gauge
+type GaugeOps interface {
+	// Set sets the Gauge to the given value.
+	Set(float64)
+	// Add(1)
```

Comment:
nit: comments should be prefixed by the method they are commenting by convention.

Reply:
yeah - let's maybe just copy comments from
https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#Gauge

Reply:
Some of the comments here are just placeholders, yes they need to follow conventions and be useful.

Reply:
I did not just copy the existing comments because we are talking about different copyright holders (for now).